packages/block-editor/src/components/inserter/hooks/use-block-type-impressions.native.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { setBlockTypeImpressionCount } from '@wordpress/react-native-bridge';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../../store';
+
+function useBlockTypeImpressions( blockTypes ) {
+	const { blockTypeImpressions, enableEditorOnboarding } = useSelect(
+		( select ) => {
+			const { getSettings: getBlockEditorSettings } = select(
+				blockEditorStore
+			);
+			const { editorOnboarding, impressions } = getBlockEditorSettings();
+
+			return {
+				blockTypeImpressions: impressions,
+				enableEditorOnboarding: editorOnboarding,
+			};
+		},
+		[]
+	);
+	const { updateSettings } = useDispatch( blockEditorStore );
+
+	const items = enableEditorOnboarding
+		? blockTypes.map( ( b ) => ( {
```

Comment:
Minor non blocking nitpick: would you consider a longer name for the element in callback ? perhaps the singular of the collection: `blockType`

Reply:
@jhnstn definitely. I intended to rename this to avoid the cryptic name, but forgot. Thank you for calling this out.